test/s3path.jl

```diff
@@ -75,6 +75,32 @@ function test_s3_readpath(p::PathSet)
     end
 end
 
+function test_s3_walkpath(p::PathSet)
+    @testset "walkpath - S3" begin
+        # Test that we still return parent prefixes even when no "directory" objects
+        # have been created by a `mkdir`, retaining consistency with `readdir`.
+        _root = p.root / "s3_walkpath/"
+
+        _foo = _root / "foo/"
+        _baz = _foo / "baz.txt"
+        _bar = _root / "bar/"
+        _qux = _bar / "qux/"
+        _quux = _qux / "quux.tar.gz"
+
+        # Only write the leaf files
+        write(_baz, read(p.baz))
+        write(_quux, read(p.quux))
+
+        topdown = [_bar, _qux, _quux, _foo, _baz]
+        bottomup = [_quux, _qux, _bar, _baz, _foo]
+
+        @test collect(walkpath(_root; topdown=true)) == topdown
+        @test collect(walkpath(_root; topdown=false)) == bottomup
```

Comment:
Would be good to have a test that uses `onerror`

Reply:
I made an issue as that change would require introducing mocking and more extensive test updates. I think a separate PR should be made to introduce mocked failure here and in several other parts of the code that does error handling.

https://github.com/JuliaCloud/AWSS3.jl/issues/180

Reply:
Thanks!